direnvrc

```diff
@@ -187,7 +187,7 @@ use_nix() {
     dump_cmd="echo _____direnv_____; \"$direnv\" dump bash"
     tmp=$("${NIX_BIN_PREFIX}nix-shell" \
       "${experimental_flags[@]}" \
-      --show-trace --pure "$@" --run "$dump_cmd")
+      --show-trace "$@" --run "$dump_cmd")
```

Comment:
I don't think we have to `--pure` otherwise we cannot cache its output since it also include the users environment.

Reply:
Is there maybe a way to save the old value of XDG_DATA_DIRS and than just append the value from nix-shell?

Reply:
I just hit this issue, should be possible to handle `XDG_DATA_DIRS`  the same way `PATH` is handled in `_nix_import_env()`?